artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -1357,7 +1357,7 @@ private PagedReference moveNext() {
                   break;
                }
 
-               int nextFileOffset = message.getPosition().getFileOffset() == -1 ? -1 : message.getPosition().getFileOffset() + message.getPagedMessage().getEncodeSize() + Page.SIZE_RECORD;
+               int nextFileOffset = message.getPosition().getFileOffset() == -1 ? -1 : message.getPosition().getFileOffset() + message.getPagedMessage().getStoredSize() + Page.SIZE_RECORD;
```

Comment:
I would encapsulate this computation as part of `PagedMessage` or `PagePosition` , possibly adding a test

Reply:
I agree, and I raised that with you. but this is not part of my change here though. this could be a different PR.